src/utilities/filetypes/WorkflowJSON.cpp

```diff
@@ -65,6 +67,10 @@ namespace detail {
 
     parseSteps();
     parseRunOptions();
+
+    if (!p.empty()) {
+      setOswPath(p, false);
+    }
```

Comment:
Meh. You can guess why it's the way it is.

* The string constructor was really only meant to take an actual JSON string in originally
* The path& one was to read from a file on disk

People using ruby especially would try to do `OpenStudio::WorkflowJSON.new("/my/workflow.osw")` (instead of `OpenStudio::WorkflowJSON.new(OpenStudio::toPath("/my/workflow.osw"))` and would be very confused, so instead the string version when it doesn't parse correctly, it falls back to check if the string argument is actually a path, and if so I'm pretty sure the intent was to do **exactly** the same thing.  C++ constructor chaining is not possible, so instead of bothering to make an internal method `fromPath(...)` for a couple of lines, the path version was copy pasted in. This happened in https://github.com/NREL/OpenStudio/commit/a7efe66c65a78924206b9e2c99af8be3422b3d73 , and probably just forgot the setOswPath since the flow control (with parseSteps etc) was different.